ansible_base/lib/utils/views/decorators.py

```diff
@@ -0,0 +1,28 @@
+from django.conf import settings
+
+oauth2_provider_installed = "ansible_base.oauth2_provider" in getattr(settings, "INSTALLED_APPS", [])
+
+
+def attach_oauth2_permission_class_to_view_class(view_class):
+    """
+    Prepend OAuth2ScopePermission to all APIVews that do not have
+    blank permission classes or AllowAny in its permission classes.
+
+    Intended to be used as a decorator for APIView classes.
+
+    :param view_class: rest_framework APIView instance
+    """
+    if oauth2_provider_installed:
+        from ansible_base.oauth2_provider.checks.permisssions_check import OAuth2ScopePermissionCheck
```

Comment:
So, I'm not sure about this for a few reasons. Generally, using `inspect` like the check does in production code is something we want to avoid. But also I feel like having a decorator here makes it less clear what is going on.

I think it is better for views and viewsets to explicitly add `OAuth2ScopePermission` to their `permission_classes`, then it's clear at a glance what is happening. Particularly around security/permissions code, I think being explicit and clear is important.

I also feel weird about using a Django "check" in production code in general. I think it's better to just catch this stuff at CI time.

Thoughts on that?

Reply:
`inspect` has been removed from the check in general now.

I've also changed the mechanism for adding `OAuth2ScopePermission` to other django apps that do not require `oauth2_provider` to be installed.